Compiler error in tests with Boost 1.67

I am trying to build the current master branch with Boost 1.67 and keep getting the following error:
`[1/2] Building CXX object test/io/CMakeFiles/test_grabbers.dir/test_grabbers.cpp.o
FAILED: test/io/CMakeFiles/test_grabbers.dir/test_grabbers.cpp.o 
[...]
../test/io/test_grabbers.cpp: In member function ‘virtual void PCL_PCDGrabber_Test::TestBody()’:
../test/io/test_grabbers.cpp:52:66: error: no matching function for call to ‘boost::date_time::subsecond_duration<boost::posix_time::time_duration, 1000000>::subsecond_duration(double)’
   boost::this_thread::sleep (boost::posix_time::microseconds (1E6));
                                                                  ^
In file included from /usr/local/include/boost/date_time/posix_time/posix_time_config.hpp:16:0,
                 from /usr/local/include/boost/date_time/posix_time/posix_time_system.hpp:13,
                 from /usr/local/include/boost/date_time/posix_time/ptime.hpp:12,
                 from /usr/local/include/boost/date_time/posix_time/posix_time_types.hpp:12,
                 from /usr/local/include/boost/thread/thread_time.hpp:11,
                 from /usr/local/include/boost/thread/lock_types.hpp:18,
                 from /usr/local/include/boost/thread/pthread/mutex.hpp:16,
                 from /usr/local/include/boost/thread/mutex.hpp:16,
                 from ../io/include/pcl/io/boost.h:49,
                 from ../io/include/pcl/io/file_io.h:43,
                 from ../io/include/pcl/io/pcd_io.h:44,
                 from ../test/io/test_grabbers.cpp:4:
/usr/local/include/boost/date_time/time_duration.hpp:285:14: note: candidate: template<class T> boost::date_time::subsecond_duration<base_duration, frac_of_second>::subsecond_duration(const T&, typename boost::enable_if<boost::is_integral<T>, void>::type*)
     explicit subsecond_duration(T const& ss,
              ^~~~~~~~~~~~~~~~~~
/usr/local/include/boost/date_time/time_duration.hpp:285:14: note:   template argument deduction/substitution failed:
/usr/local/include/boost/date_time/time_duration.hpp: In substitution of ‘template<class T> boost::date_time::subsecond_duration<boost::posix_time::time_duration, 1000000>::subsecond_duration(const T&, typename boost::enable_if<boost::is_integral<T>, void>::type*) [with T = double]’:
../test/io/test_grabbers.cpp:52:66:   required from here
/usr/local/include/boost/date_time/time_duration.hpp:285:14: error: no type named ‘type’ in ‘struct boost::enable_if<boost::is_integral<double>, void>’
In file included from /usr/local/include/boost/date_time/posix_time/posix_time_config.hpp:16:0,
                 from /usr/local/include/boost/date_time/posix_time/posix_time_system.hpp:13,
                 from /usr/local/include/boost/date_time/posix_time/ptime.hpp:12,
                 from /usr/local/include/boost/date_time/posix_time/posix_time_types.hpp:12,
                 from /usr/local/include/boost/thread/thread_time.hpp:11,
                 from /usr/local/include/boost/thread/lock_types.hpp:18,
                 from /usr/local/include/boost/thread/pthread/mutex.hpp:16,
                 from /usr/local/include/boost/thread/mutex.hpp:16,
                 from ../io/include/pcl/io/boost.h:49,
                 from ../io/include/pcl/io/file_io.h:43,
                 from ../io/include/pcl/io/pcd_io.h:44,
                 from ../test/io/test_grabbers.cpp:4:
/usr/local/include/boost/date_time/time_duration.hpp:270:30: note: candidate: boost::date_time::subsecond_duration<boost::posix_time::time_duration, 1000000>::subsecond_duration(const boost::date_time::subsecond_duration<boost::posix_time::time_duration, 1000000>&)
   class BOOST_SYMBOL_VISIBLE subsecond_duration : public base_duration
                              ^~~~~~~~~~~~~~~~~~
/usr/local/include/boost/date_time/time_duration.hpp:270:30: note:   no known conversion for argument 1 from ‘double’ to ‘const boost::date_time::subsecond_duration<boost::posix_time::time_duration, 1000000>&’
/usr/local/include/boost/date_time/time_duration.hpp:270:30: note: candidate: boost::date_time::subsecond_duration<boost::posix_time::time_duration, 1000000>::subsecond_duration(boost::date_time::subsecond_duration<boost::posix_time::time_duration, 1000000>&&)
/usr/local/include/boost/date_time/time_duration.hpp:270:30: note:   no known conversion for argument 1 from ‘double’ to ‘boost::date_time::subsecond_duration<boost::posix_time::time_duration, 1000000>&&`


So far I have not found any similar issue here for PCL, but I think there are a few for different projects (e.g. https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=227540).

Anybody got an idea on how to fix this? Or am I doing something wrong?

Btw. for the time being I disabled the tests and do not get build errors anymore.

--- a/test/io/test_grabbers.cpp
+++ b/test/io/test_grabbers.cpp
@@ -49,7 +49,7 @@ TEST (PCL, PCDGrabber)
   grabber.registerCallback (fxn);
   grabber.start ();
   // 1 second should be /plenty/ of time
-  boost::this_thread::sleep (boost::posix_time::microseconds (1E6));
+  boost::this_thread::sleep (boost::posix_time::seconds (1));
   grabber.stop ();
 
   //// Make sure they match

